crates/biome_js_analyze/src/lint/nursery/no_label_without_control.rs

```diff
@@ -0,0 +1,290 @@
+    ///
+    /// There are two supported ways to associate a label with a control:
+    /// - Wrapping a control in a label element.
+    /// - Adding a `for` attribute (or `htmlFor` in React) to a label and assigning it a DOM ID string that indicates an input on the page.
+    ///
+    /// This rule checks that any `label` element (or an indicated custom component that will output a `label` element) meets one of this conditions:
+    /// - Wraps an `input` element (or an indicated custom component that will output an `input` element)
+    /// - Has a `for` or `htmlFor` attribute and that the `label` element/component has accessible text content.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <label for="js_id" />;
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <label for="js_id"><input /></label>;
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <label htmlFor="js_id" />;
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <label htmlFor="js_id"><input /></label>;
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <label>A label</label>;
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <div><label /><input /></div>;
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```jsx
+    /// <label for="js_id" aria-label="A label" />;
+    /// ```
+    ///
+    /// ```jsx
+    /// <label for="js_id" aria-labelledby="A label" />;
+    /// ```
+    ///
+    /// ```jsx
+    /// <label htmlFor="js_id" aria-label="A label" />;
+    /// ```
+    ///
+    /// ```jsx
+    /// <label htmlFor="js_id" aria-labelledby="A label" />;
+    /// ```
+    ///
+    /// ```jsx
+    /// <label>A label<input /></label>;
+    /// ```
+    ///
+    /// ```jsx
+    /// <label>A label<textarea /></label>;
+    /// ```
+    ///
+    /// ```jsx
+    /// <label><img alt="A label" /><input /></label>;
+    /// ```
+    ///
+    /// ## Options
+    ///
+    /// The rule supports the following options:
+    /// - `controlComponents` - An array of component names that should be considered the same as an `input` element.
+    /// - `labelAttributes` - An array of attributes that should be treated as the `label` accessible text content.
+    /// - `labelComponents` - An array of component names that should be considered the same as a `label` element.
+    ///
+    /// ```json
+    /// {
+    ///     "//": "...",
+    ///     "options": {
+    ///         "controlComponents": ["CustomInput"],
+    ///         "labelAttributes": ["label"],
+    ///         "labelComponents": ["CustomLabel"]
+    ///     }
+    /// }
+    /// ```
+    ///
+    pub NoLabelWithoutControl {
+        version: "next",
+        name: "noLabelWithoutControl",
+        language: "jsx",
+        sources: &[RuleSource::EslintJsxA11y("label-has-associated-control")],
+        recommended: false,
```

Comment:
I think it should be recommended, what do you think?

Reply:
I think it should, but when I was implementing `noUselessConcat` @arendjr mentioned that nursery rules are not recommended by default, just when they get promoted to something else

Reply:
They are not, but it's a change that occurs at compile time, so you don't need to worry. It's fine to have them recommended in the source code.

Reply:
oh, I see, changed it to recommended then 😃